src/main/java/spoon/decompiler/Decompiler.java

```diff
@@ -0,0 +1,16 @@
+package spoon.decompiler;
+
+import java.io.File;
+
+public interface Decompiler {
+
+	/**
+	 * Sets the output directory for source generated.
+	 *
+	 * @param jar
+	 * 		Path to jar to be analyzed.
+	 * @param outputDir
+	 * 		Path for the output directory for decompilation.
+	 */
+	void decompile(File jar, File outputDir);
```

Comment:
What about

 	List<SpoonFile> decompile(String jarPath); 
 	List<SpoonResource> decompile(String jarPath); 
 	CtModel decompile(String jarPath);

Reply:
I'm not convinced by either of this solutions. My plan was to call the decompiler only if the input jar has been modified after last decompilation or if user forces it. I would prefer having directory declaration in JarLauncher than in Decompiler.